spec/support/file_system.rb

```diff
@@ -43,6 +37,23 @@ def chdir_and_allow_nesting(path)
       Dir.chdir(original_path)
     end
   end
+
+  private
+
+  def switch_home_directory(&block)
```

Comment:
How about considering using a library for this? https://github.com/defunkt/fakefs